### PR TITLE
client/next: inits next app with typescript

### PR DESCRIPTION
**Pull request by grantkang5**
## Summary
Inits next app with typescript and tailwind

## How did you test your changes?
*Wrote unit tests, tested changes locally*

### CODE REVIEW

File: client/.gitignore [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+yarn-error.log*
+
+# local env files
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+# vercel
+.vercel
```
Comment (jeremyhcho) [inline client/.gitignore:34]: what kind of config does vercel have that we want to gitignore?

File: client/pages/_app.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import 'tailwindcss/tailwind.css'
+
+function MyApp({ Component, pageProps }) {
```
Comment (jeremyhcho) [inline client/pages/_app.js]: nit: fat arrow declaration?

File: client/pages/api/hello.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+
+export default function helloAPI(req, res) {
```
Comment (jeremyhcho) [inline client/pages/api/hello.js]: nit: fat arrow?

File: client/pages/_app.js (hunk 2 above)
Comment (grantkang5) [reply client/pages/_app.js]: Oops! Project was not created with typescript version - fixed.

File: client/.gitignore (hunk 1 above)
Comment (grantkang5) [reply client/.gitignore:34]: This is the main config file for the vercel cli to point to the proper vercel instance. Recommended to ignore, not certain we will use vercel yet.

File: client/styles/Home.module.css [hunk 4]
```diff
@@ -0,0 +1,121 @@
+.container {
```
Comment (jeremyhcho) [inline client/styles/Home.module.css:1]: wat dis?

Comment (jeremyhcho) [reply client/styles/Home.module.css:1]: skeleton?

File: client/styles/Home.module.css [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  min-height: 100vh;
+  padding: 0 0.5rem;
+  display: flex;
```
Comment (jeremyhcho) [inline client/styles/Home.module.css:4]: i notice this file is in `client/styles` and being referenced by `client/pages`

does that mean all css files will be in a separate subdirectory than components?

File: client/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
+{
+  "compilerOptions": {
+    "target": "es5",
```
Comment (jeremyhcho) [inline client/tsconfig.json:3]: why not es6?

https://www.typescriptlang.org/tsconfig#target

**Review by jeremyhcho — APPROVED** [review]
69

File: client/styles/Home.module.css (hunk 4 above)
Comment (grantkang5) [reply client/styles/Home.module.css:1]: Boilerplate code. Keep for now for @mkshin1 to reference.

File: client/styles/Home.module.css (hunk 5 above)
Comment (grantkang5) [reply client/styles/Home.module.css:4]: Yes. Next js is using css modules out of the box and we can alias `styles` directory if need be. I prefer styles to be in its own directory tbh but most likely we will not need any of this if we fully commit to tailwind.

File: client/tsconfig.json (hunk 6 above)
Comment (grantkang5) [reply client/tsconfig.json:3]: es6 less browser support. No added benefit in code language.

**Comment by grantkang5** [comment]
> 69

70

File: client/styles/Home.module.css (hunk 5 above)
Comment (mkshin1) [reply client/styles/Home.module.css:4]: tailwind (=

File: client/styles/Home.module.css (hunk 4 above)
Comment (mkshin1) [reply client/styles/Home.module.css:1]: ty

**Review by mkshin1 — APPROVED** [review]

**Review by mkshin1 — APPROVED** [review]

**Review by mkshin1 — APPROVED** [review]

**Review by mkshin1 — APPROVED** [review]

**Pull request merged**